boa_engine/src/builtins/temporal/duration/record.rs

```diff
@@ -0,0 +1,1810 @@
+use boa_macros::utf16;
+
+use crate::{
+    builtins::{
+        options::RoundingMode,
+        temporal::{self, create_temporal_date, options::TemporalUnit, to_temporal_date},
+    },
+    js_string, Context, JsNativeError, JsObject, JsResult, JsValue,
+};
+
+use super::super::{calendar, to_integer_if_integral, zoned_date_time};
+
+// ==== `DateDuration` ====
+
+#[derive(Debug, Default, Clone, Copy)]
+pub(crate) struct DateDuration {
+    years: f64,
+    months: f64,
+    weeks: f64,
+    days: f64,
```

Comment:
Can you add a link to https://tc39.es/proposal-temporal/#sec-properties-of-temporal-duration-instances here and in `TimeDuration/DateTimeDuration`? It would be good to document that we're not using integers because the spec allows all float64-representable integers as valid values, and `f64` has many integers that aren't representable by `i64`.